// persistence/idManager.js
var Datastore = require("nedb");
var spromise  = require("spromise");

var ready = spromise.defer();
var db = {}, companyID = 1, peopleID = 1, positionID =1;

db.Ids = new Datastore({
  filename: "storage/ids.db",
  autoload: true,
  onload: function(err) {
    if ( err ) {
      ready.reject(err);
    }
    else {
      db.Ids.findOne({"companyID": "seed"}, function(err, result) {
        if ( result && result.value ) {
          companyID = result.value;
        }
        else {
          db.Ids.insert({"value": companyID, "companyID": "seed"}, function(){});
        }

        ready.resolve(db);
      });
     db.Ids.findOne({"peopleID": "seed"}, function(err, result) {
        if ( result && result.value ) {
          peopleID = result.value;
        }
        else {
          db.Ids.insert({"value": peopleID, "peopleID": "seed"}, function(){});
        }

        ready.resolve(db);
      });
     db.Ids.findOne({"positionID": "seed"}, function(err, result) {
        if ( result && result.value ) {
          positionID = result.value;
        }
        else {
          db.Ids.insert({"value": positionID, "positionID": "seed"}, function(){});
        }

        ready.resolve(db);
      });
    }
  }
});

db.Ids.loadDatabase();

function generateCompanyId() {
  companyID++;
  db.Ids.update({"companyID": "seed"}, { $set: { "value": companyID } }, {}, function(){});
  return companyID;
}

function generatePeopleId() {
  peopleID++;
  db.Ids.update({"peopleID": "seed"}, { $set: { "value": peopleID } }, {}, function(){});
  return peopleID;
}

function generatePositionId() {
  positionID++;
  db.Ids.update({"positionID": "seed"}, { $set: { "value": positionID } }, {}, function(){});
  return positionID;
}

module.exports = {
  ready: ready.promise,
  generateCompanyId: generateCompanyId,
  generatePeopleId: generatePeopleId,
  generatePositionId: generatePositionId
};
