// services/jobSeeker.js
var extender    = require("utils/extender");
var md5         = require("utils/md5");
var _           = require("lodash");
var spromise    = require("spromise");
var companies   = require("persistence/companies");
var people      = require("persistence/people");
var positions   = require("persistence/positions");
var idManager   = require("persistence/idManager");


var ready = spromise.when(idManager.ready);


function jobSeeker() {
  this.initialize();
}


function initialize(options) {
  options     = options || {};
  this.active = {};
  this.ready  = ready;
}

function company(data) {
    return spromise(function(resolve){
        var _msg = {
            "companyName": data.companyName,
            "companyPhone": data.companyPhone,
            "companyAddress": data.companyAddress,
            "companyID": idManager.generateCompanyId(),
        };

        companies.insert(_msg);
        resolve();
    });
}

function companylist() {
    return spromise(function(resolve){
    var query = {};

    companies.find(query, function(err, companies) {
      resolve(companies);
    });
  });
}

module.exports = extender.extend(jobSeeker, {
  initialize: initialize,
  company: company,
  companylist: companylist
});
